packages/react-native-web/src/exports/Image/index.js

```diff
@@ -332,24 +340,94 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+/**
+ * This component handles specifically loading an image source with header
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore
+  const nextSource: SourceObject = props.source;
+  const prevSource = React.useRef<SourceObject>({});
+  const cleanup = React.useRef<Function>(() => {});
+  const [blobUri, setBlobUri] = React.useState('');
+
+  const { onError, onLoadStart } = props;
+
+  React.useEffect(() => {
+    if (!hasSourceDiff(nextSource, prevSource.current)) return;
+
+    // When source changes we want to clean up any old/running requests
+    cleanup.current();
+
+    prevSource.current = nextSource;
+
+    let uri;
+    const abortCtrl = new AbortController();
+    const request = new Request(nextSource.uri, {
+      headers: nextSource.headers,
+      signal: abortCtrl.signal
+    });
+    request.headers.append('accept', 'image/*');
+
+    if (onLoadStart) onLoadStart();
+
+    fetch(request)
+      .then((response) => response.blob())
+      .then((blob) => {
+        uri = URL.createObjectURL(blob);
+        setBlobUri(uri);
+      })
+      .catch((error) => {
+        if (error.name !== 'AbortError' && onError) {
+          onError({ nativeEvent: error.message });
+        }
+      });
+
+    // Capture a cleanup function for the current request
+    // The reason for using a Ref is to avoid making this function a dependency
+    // Because the change of a dependency would otherwise would re-trigger a hook
```

Comment:
```suggestion
    // Capture a cleanup function for the current request.
    // The reason for using a Ref is to avoid making this function a dependency
    // because the change of a dependency would otherwise re-trigger a hook.
```